restic/kubernetes/pod_exec.go

```diff
@@ -73,6 +74,8 @@ func PodExec(pod BackupPod, log logr.Logger) (*ExecData, error) {
 
 		if err != nil {
 			execLogger.Error(err, "streaming data failed", "namespace", pod.Namespace, "pod", pod.PodName)
+			// we just completely hard fail the whole backup pod
+			os.Exit(1)
```

Comment:
why don't You simply return err to some new resChannel and then return it in the end of a function? os.Ecit(1) sounds bit harsh